internal/oci/options.go

```diff
@@ -0,0 +1,57 @@
+package oci
+
+import (
+	"go.flipt.io/flipt/internal/containers"
+	"go.flipt.io/flipt/internal/oci/ecr"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// StoreOptions are used to configure call to NewStore
+// This shouldn't be handled directory, instead use one of the function options
+// e.g. WithBundleDir or WithCredentials
+type StoreOptions struct {
+	bundleDir       string
+	manifestVersion oras.PackManifestVersion
+	auth            credentialFunc
+}
+
+// WithCredentials configures username and password credentials used for authenticating
+// with remote registries
+func WithCredentials(kind, user, pass string) containers.Option[StoreOptions] {
+	switch kind {
+	case "aws-ecr":
+		return WithAWSECRCredentials()
+	default:
+		return WithStaticCredentials(user, pass)
+	}
+}
```

Comment:
Re-adding this (not sure what I did wrong there):

We should pop some validation somewhere on this `kind` or oci type parameter.
Just so people don't typo it, then it string static and they get a misleading error.